include/benchmark/benchmark.h

```diff
@@ -544,8 +506,7 @@ inline BENCHMARK_ALWAYS_INLINE void DoNotOptimize(Tp&& value) {
   asm volatile("" : "+m,r"(value) : : "memory");
 #endif
 }
-#endif
-#elif defined(BENCHMARK_HAS_CXX11) && (__GNUC__ >= 5)
+#elif (__GNUC__ >= 5)
```

Comment:
This needs a comment. It took me a while to get that it is for `#if !defined(__GNUC__) || defined(__llvm__) || defined(__INTEL_COMPILER)`

Reply:
it took me ages to navigate these ifdefs too.  added comments.